app/src/main/java/com/globant/example/mentorapp/subscriberDetails/presentation/view/fragment/UserDetailsFragment.java

```diff
@@ -1,29 +1,186 @@
+        }
+
+        listReposRecyclerView.setAdapter(listRepoAdapter);
+        return view;
+    }
+
+    @Override
+    public void render(UserDetailsViewModel detailsViewModel) {
+        if (detailsViewModel.getUserDetails() != null) {
+            model.setUsersDetails(detailsViewModel.getUserDetails());
+            setUserInformation(detailsViewModel.getUserDetails());
+        }
+        if (detailsViewModel.getUserRepos() != null) {
+            model.setUserRepos(detailsViewModel.getUserRepos());
+            totalRepositories.setText(
+                    String.format(getActivity().getString(R.string.tv_detail_repositories), detailsViewModel.getUserRepos().size()));
+            usersRepositoryListReady();
+        }
+        if (detailsViewModel.getError() != null) {
+            showErrorMessage(detailsViewModel.getError());
+        }
+        if (detailsViewModel.getProgress()) {
+            parent.showProgress();
+        } else {
+            parent.hideProgress();
+        }
+    }
+
+    private void subscribeRepoList() {
+        final Observer<List<RepositoryModel>> usersObserver = new Observer<List<RepositoryModel>>() {
+            @Override
+            public void onChanged(@Nullable List<RepositoryModel> repositoryViewModelList) {
+                listRepoAdapter = new UserRepositoriesAdapter(repositoryViewModelList);
+                usersRepositoryListReady();
+            }
+        };
+        model.getUserRepos().observe(this, usersObserver);
     }
 
+    private void subscribeUserDetails() {
+        final Observer<UserDetailsModel> usersObserver = new Observer<UserDetailsModel>() {
+            @Override
+            public void onChanged(@Nullable UserDetailsModel detailsModel) {
+                setUserInformation(detailsModel);
+            }
+        };
+        model.getUserDetails().observe(this, usersObserver);
+    }
+
+    private void setUserInformation(UserDetailsModel userDetails) {
+        userName.setText(
+                String.format(getActivity().getString(R.string.tv_detail_user_name), userDetails.getUserName()));
+        userLocation.setText(
+                String.format(getActivity().getString(R.string.tv_detail_location), userDetails.getLocation()));
+        company.setText(
+                String.format(getActivity().getString(R.string.tv_detail_company), userDetails.getCompany()));
+        followers.setText(
+                String.format(getActivity().getString(R.string.tv_detail_followers), userDetails.getFollowers()));
+        following.setText(
+                String.format(getActivity().getString(R.string.tv_detail_following), userDetails.getFollowing()));
+
+        Picasso.with(getActivity()).load(userDetails.getImageUrl())
+                .error(R.drawable.ic_not_found)
+                .placeholder(R.drawable.ic_placeholder)
+                .into(userImageView);
+    }
+
+    private void usersRepositoryListReady() {
+        listReposRecyclerView.addItemDecoration(
```

Comment:
You can execute the blocked between lines 160 to 164 just once after initializing listReposRecyclerView object.